### ISSUE TITLE
sympy.Sum() bug

### ISSUE THREAD

**Issue by yz3558** [sympy/sympy#23156]
x=sympy.Symbol("x")
a=sympy.Symbol("a")
sympy.Sum(1/sympy.gamma(x),(x,0,0)).doit()
sympy.Sum(1/sympy.gamma(x),(x,0,a)).doit().subs({a:0})

These two lines give different results. The first one gives 0, while the second one gives e.
The first result is correct.

**Comment by anutosh491** [comment]
Well I can say that the result for a random integer `a` is correct 
```
>>> Sum(1/gamma(x), (x, 0, a)).doit()
-E*a*lowergamma(a, 1)/gamma(a + 1) + E
```

**Comment by anutosh491** [comment]
I can confirm that all cases involving positive `a` look fine but doesn't work otherwise . So does it mean the general sum should only be returned if `a` is positive .
```
>>> Sum(1/gamma(x), (x, 0, a)).doit().subs(a, -2)
E
```
Yeah I would expect `nan` as pointed above but otherwise can't comment just by surface level analysis , cause the general sum till a random positive integer `a` looks fine to me !
Obviously the code snippet answering `Sum(1/gamma(x), (x, 0, 0)).doit()` is as follows so it returns `1/zoo` 
```
def eval_sum(f, limits):
    (i, a, b) = limits
    if a == b:
        return f.subs(i, a)
```

**Comment by yz3558** [comment]
Can we define `0*lowergamma(0,1)=1`? This will make it consistent between 0 and positive `a` values.

**Comment by anutosh491** [comment]
> Can we define `0*lowergamma(0,1)=1`? This will make it consistent between `a=0` and positive `a` values.

I don't think we can do this cause `lowergamma(0, 1)` evaluates `oo` and as per sympy convention `0*oo = nan`

**Comment by yz3558** [comment]
If we identify a lowergamma(0,1) in the expression, we can always see if we have a 0 multiplies this term. If so, we get a 1. I think it is better nan.

**Comment by anutosh491** [comment]
> If we identify a`lowergamma(0,1)` in the expression, instead of converting it to `oo` first, can we first see if we have a `0` multiplies this term? If so, we get a `1`.

I don't think this would be the best way to address this . We shouldn't be writing code to address a particular cases explicitly and try to buid an approach which generally works fine overall !
